### DISCUSSION TITLE
Dúvida em relação a hospedagem compartilhada, VPS e linguagens de programação

### DISCUSSION THREAD

**Discussion by hmslima** [backend-br/forum#122] [Q&A] [answered]
Tradicionalmente, o *back end* era feito através da linguagem PHP e suas frameworks, mas hoje em dia temos o NodeJS que está ganhando espaço e o próprio Java que sempre esteve por aí.

Iniciei o curso de ADS e meu foco é em me tornar um desenvolvedor *back end*, portanto naturalmente que comecei a estudar JavaScrit(NodeJS) e Java. Eu pensava em, no futuro, assim que eu tivesse um mínimo de domínio, criar meus projetinhos e contratar um serviço de hospedagem de sites para disponibilizá-los. Foi aí que aprendi que há dois tipos de serviços: **hospedagem de sites** _(ou **hospedagem compartilhada**)_ e **VPS**, e que a tradicional hospedagem de sites é bem mais limitada em relação ao VPS, [um exemplo é o Hostgator](https://suporte.hostgator.com.br/hc/pt-br/articles/115000384693).

Eu até entendo que para coisas envolvendo linguagens como C/C++, você precisa de maior domínio sobre a máquina *(coisa que o VPS oferece)*, mas foi um choque pra mim perceber que até tecnologias como NodeJS, Java e PostgreSQL estão disponíveis apenas – pelo menos é assim na Hostgator – em VPS!

1. Quer dizer então que se eu tiver condições de contratar apenas um serviço de hospedagem compartilhada, devo me limitar ao PHP *– que é uma linguagem que pouco a pouco perdendo espaço para o NodeJS e Java –*, ao Ruby on Rails e ao Python? *Sei que PHP, Ruby e Python são grandes tecnologias, especialmente este último, mas é que meu foco todo estava no JavaScript e Java por conta da empregabilidade...*

2. A propósito, por que PHP, Ruby (on Rails) e Python *(e MySQL)* estão disponíveis em serviços de hospedagem compartilhada, mas o NodeJS e o Java *(e o PostgreSQL)* não?

3. Há previsão de, no futuro, trazerem tecnologias  como NodeJS e PostgreSQL para a hospedagem compartilhada? Não falo do Java porque esta tecnologia está aí há anos e isso aparentemente nunca aconteceu.

**Comment by reginaldojunior** [comment, marked as answer]
Opa @hmslima!

Sim, alguns anos atrás a hospedagem compartilhada dominava a internet e muitos sites utilizam este tipo de serviço para disponibilizar seus serviços. Hoje apesar de existir vejo muito poucas pessoas utilizando esse meio para disponibilizar seus projetos, a tendencia com o avanço do docker, kubernetes e a cultura devops assim como o aumento de profissionais da área é que seja mais utilizado esses tipos de serviço como VPS. 

Com o VPS você além de ter acesso ao ssh da máquina tem muito mais liberdade para configurar o servidor da forma que você desejar, apesar de ser um passo a mais de aprender, como configurar um servidor, e etc... A maior parte das máquinas já vem com um nginx ou apache instalado, se não for o caso, recomendo estudar um pouco sobre estes servidores.

Sobre o PHP antes ter essa facilidade, realmente não sei dizer o real motivo, mas tenho umas suspeitas, o php por ter ficado muito famoso pela sua facilidade em escrever projetos web na decada de 90 e anos 2000, perdendo um pouco de força agora nos anos 10, e 20 para outras linguagens tão boas quanto. Facilitou com que houvesse padrões de servidores e esses ficassem disponiveis para que os web designs da epóca pudessem disponibilizar seu projetos, hoje com a variedade de tecnologia e stacks fica mais dificil e mais facil disponibilizar a máquina para que o programador configure da melhor forma.

Outros serviços que indico são os proprio github-pages para páginas javascript e mais voltado ao backend o digitalocean e o heroku arrebenta e tem ótimos planos gratuitos.

Espero ter ajudado.

**Reply by hmslima** [reply]
Obrigado por responder.
